lib/src/iter.rs

```diff
@@ -243,6 +252,16 @@ impl<'r> RefsReferringTo<'r> {
         }
         Ok(())
     }
+
+    /// Preload the iterator with references
+    ///
+    /// The references supplied will be returned by the iterator.
+    ///
+    pub fn preload<I>(&mut self, references: I)
+        where I: IntoIterator<Item = git2::Reference<'r>>
+    {
+        self.current_refs.extend(references);
+    }
```

Comment:
We cold also implement `Extend`, which would spare us a pointless collection within the GC. But I felt it would be wrong, semantically. What's your opinion, @matthiasbeyer?

Reply:
I think it is rather fitting semantically. I'm not sure what you mean by

>  spare us a pointless collection within the GC

though.

Reply:
https://github.com/neithernut/git-dit/blob/9c039bba0803029f79db532e3276f56603245366/lib/src/gc.rs#L143